app/models/user.rb

```diff
@@ -72,5 +72,9 @@ def email=(value)
   def name
     "#{first_name} #{last_name}"
   end
- 
+
+  def ensure_district(district:)
```

Comment:
* last time I check a method that mutates does not necessarily have to be a bang method, I thought it was intended for `danger` methods and can't really see what's makes this method dangerous.
* I thought `ensure` in the prefix would be explicit enough and it's in fact adding the district only when the user doesn't have the district added to it.
* the name you suggest `conditionally_add_district` doesn't really tell me what's the condition is and it seems to be leaking part of the implementation in the name of the method.